test-tests/shared/src/test/scala/zio/test/TestAspectSpec.scala

```diff
@@ -0,0 +1,108 @@
+              assertM(ref.get, equalTo(1))
+            } @@ around(ref.set(1), ref.set(-1))
+            result <- succeeded(spec)
+            after  <- ref.get
+          } yield assert(result && (after == -1), isTrue)
+        },
+        testM("js applies test aspect only on ScalaJS") {
+          for {
+            ref    <- Ref.make(false)
+            spec   = test("test")(assert(true, isTrue)) @@ js(after(ref.set(true)))
+            _      <- execute(spec)
+            result <- ref.get
+          } yield if (TestPlatform.isJS) assert(result, isTrue) else assert(result, isFalse)
+        },
+        testM("jsOnly runs tests only on ScalaJS") {
+          val spec   = test("Javascript-only")(assert(TestPlatform.isJS, isTrue)) @@ jsOnly
+          val result = if (TestPlatform.isJS) succeeded(spec) else ignored(spec)
+          result.map(assert(_, isTrue))
+        },
+        testM("jvm applies test aspect only on jvm") {
+          for {
+            ref    <- Ref.make(false)
+            spec   = test("test")(assert(true, isTrue)) @@ jvm(after(ref.set(true)))
+            _      <- execute(spec)
+            result <- ref.get
+          } yield assert(if (TestPlatform.isJVM) result else !result, isTrue)
+        },
+        testM("jvmOnly runs tests only on the JVM") {
+          val spec   = test("JVM-only")(assert(TestPlatform.isJVM, isTrue)) @@ jvmOnly
+          val result = if (TestPlatform.isJVM) succeeded(spec) else ignored(spec)
+          result.map(assert(_, isTrue))
+        },
+        testM("failure makes a test pass if the result was a failure") {
+          val spec   = test("test")(assert(throw new java.lang.Exception("boom"), isFalse)) @@ failure
+          val result = succeeded(spec)
+          result.map(assert(_, isTrue))
+        },
+        testM("failure makes a test pass if it died with a specified failure") {
+          val spec = test("test")(assert(throw new NullPointerException(), isFalse)) @@ failure(
+            TestAspectSpecHelper.failsWithException[NullPointerException]
+          )
+          succeeded(spec).map(assert(_, isTrue))
+        },
+        testM("failure does not make a test pass if it failed with an unexpected exception") {
+          val spec = test("test")(
+            assert(throw new NullPointerException(), isFalse)
+          ) @@ failure(TestAspectSpecHelper.failsWithException[IllegalArgumentException])
+          failed(spec).map(assert(_, isTrue))
+        },
+        testM("failure does not make a test pass if the specified failure does not match") {
+          val spec = test("test")(assert(throw new RuntimeException(), isFalse)) @@ failure(
+            isCase[TestFailure[String], Cause[String]]("Runtime", {
+              case TestFailure.Runtime(e) => Some(e); case _ => None
+            }, equalTo(Cause.fail("boom")))
+          )
+          failed(spec).map(assert(_, isTrue))
+        },
+        testM("failure makes tests pass on any assertion failure") {
```

Comment:
I think for this one you don't need to create a separate test and execute it. Your test could just be something like:

```scala
test("failure makes tests pass on any assertion failure) {
  assert(true, isFalse)
} @@ failure
```

Reply:
Everything made a lot more sense with this!